cmdstanpy/stanfit/laplace.py

```diff
@@ -89,16 +86,10 @@ def _assemble_draws(self) -> None:
         if self._draws.shape != (0,):
             return
 
-        with open(self._runset.csv_files[0], 'r') as fd:
-            while (fd.readline()).startswith("#"):
-                pass
-            self._draws = np.loadtxt(
-                fd,
-                dtype=float,
-                ndmin=2,
-                delimiter=',',
-                comments="#",
-            )
+        *_, draws = stancsv.parse_comments_header_and_draws(
+            self._runset.csv_files[0]
+        )
+        self._draws = stancsv.csv_bytes_list_to_numpy(draws)
```

Comment:
Not immediately sure if these functions would ever raise for GQs but if so we should wrap in a `try`/re-raise like we do in MCMC, to at least identify which chain was the problem.

We might want to do this in the other algorithms, though because they're single files it probably doesn't add as much compared to just letting the underlying error propagate -- thoughts?